aptos-move/framework/aptos-stdlib/sources/cryptography/single_key.move

```diff
@@ -0,0 +1,88 @@
+module aptos_std::single_key {
+    use aptos_std::ed25519;
+    use std::hash;
+    //
+    // Error codes
+    //
+
+    /// Wrong number of bytes were given as input when deserializing an Ed25519 public key.
+    const E_INVALID_PUBLIC_KEY_TYPE: u64 = 1;
+
+    /// Failed to deserialize the public key.
+    const E_FAILED_TO_DESERIALIZE: u64 = 2;
+
+    /// Wrong number of bytes were given as input when deserializing an Ed25519 signature.
+    const E_INVALID_SIGNATURE_SCHEME: u64 = 3;
+
+    //
+    // Constants
+    //
+
+    /// The identifier of the Single Key signature scheme, which is used when deriving Aptos authentication keys by hashing
+    /// it together with an Single Key public key.
+    const SIGNATURE_SCHEME_ID: u8 = 2;
+
+    /// Scheme identifier for Ed25519 single keys.
+    const ED25519_PUBLIC_KEY_TYPE: u8 = 0;
+
+    /// Scheme identifier for SECP256K1 single keys.
+    const SECP256K1_PUBLIC_KEY_TYPE: u8 = 1;
+
+    /// Scheme identifier for WebAuthn single keys.
+    const WEB_AUTHN_PUBLIC_KEY_TYPE: u8 = 2;
+
+    /// Scheme identifier for Keyless single keys.
+    const KEYLESS_PUBLIC_KEY_TYPE: u8 = 3;
+
+    /// Scheme identifier for Federated Keyless single keys.
+    const FEDERATED_KEYLESS_PUBLIC_KEY_TYPE: u8 = 4;
+
+    //
+    // Structs
+    //
+
+    /// An *unvalidated* any public key: not necessarily an elliptic curve point, just a sequence of 32 bytes
+    struct UnvalidatedPublicKey has copy, drop, store {
+        bytes: vector<u8>
+    }
+
+    //
+    // Functions
+    //
+
+    /// Parses the input bytes as an *unvalidated* single key.  It does check that the first byte is a valid scheme identifier.
+    public fun new_unvalidated_public_key_from_bytes(bytes: vector<u8>): UnvalidatedPublicKey {
+        let first_byte = bytes[0];
+        assert!(first_byte <= 4, std::error::invalid_argument(E_INVALID_PUBLIC_KEY_TYPE));
```

Comment:
4 is a magic number. Is there a way to share the scheme identifier enum somewhere between here and rust? Or at least have it in both and wrap in IF ELSE comments for automatic update reminders